eternalcore-api/src/main/java/com/eternalcode/core/delay/Delay.java

```diff
@@ -1,50 +1,61 @@
 package com.eternalcode.core.delay;
 
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
-
 import java.time.Duration;
-import java.time.Instant;
-import java.util.function.Supplier;
-
-public class Delay<T> {
-
-    private final Cache<T, Instant> delays;
 
-    private final Supplier<Duration> delaySettings;
-
-    public Delay(Supplier<Duration> delayProvider) {
-        this.delaySettings = delayProvider;
-
-        this.delays = CacheBuilder.newBuilder()
-            .expireAfterWrite(delayProvider.get())
-            .build();
+/**
+ * Factory class for creating delay instances.
+ * <p>
+ * Naming convention:
+ * - withDefault(...) -> DefaultDelay (uses predefined Duration)
+ * - explicit(...) -> ExplicitDelay (requires explicit Duration each time)
+ */
+public final class Delay {
+
+    private Delay() {
+        throw new UnsupportedOperationException("Utility class - do not instantiate");
```

Comment:
```suggestion
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
```
I think it would be good practice to standardize all such messages in utility classes, because currently each one looks different or is missing.